.github/workflows/install-shared-dependencies/action.yml

```diff
@@ -35,26 +39,31 @@ runs:
               brew upgrade || true
               brew install git gcc pkgconfig openssl redis coreutils
 
-        - name: Install software dependencies for Ubuntu
+        - name: Install software dependencies for Ubuntu GNU
           shell: bash
-          if: "${{ inputs.os == 'ubuntu' }}"
+          if: "${{ inputs.os == 'ubuntu' && inputs.target != 'x86_64-unknown-linux-musl' && inputs.target != 'aarch64-unknown-linux-musl'}}"
           run: |
               sudo apt update -y
               sudo apt install -y git gcc pkg-config openssl libssl-dev
+          
+        - name: Install software dependencies for Ubuntu MUSL
+          shell: bash
+          if: "${{ inputs.target == 'x86_64-unknown-linux-musl' || inputs.target == 'aarch64-unknown-linux-musl'}}"
+          run: |
+              apk update
+              wget -O - https://sh.rustup.rs | sh -s -- -y
+              source "$HOME/.cargo/env"
+              apk add protobuf-dev musl-dev make gcc redis envsubst
 
         - name: Install software dependencies for Amazon-Linux
           shell: bash
           if: "${{ inputs.os == 'amazon-linux' }}"
           run: |
               yum install -y gcc pkgconfig openssl openssl-devel which curl redis6 gettext --allowerasing
 
-        - name: Install Rust toolchain
-          uses: dtolnay/rust-toolchain@stable
-          with:
-              targets: ${{ inputs.target }}
-
-        - name: Install protoc (protobuf)
-          uses: arduino/setup-protoc@v3
+        - name: Install Rust toolchain and protoc
+          if: "${{ inputs.target != 'x86_64-unknown-linux-musl' && inputs.target != 'aarch64-unknown-linux-musl' }}"
```

Comment:
Why you don't want to run this on `musl` too?

Reply:
Unfortunately it is not supported for musl, we install those manually